api/src/test/java/org/pmiops/workbench/access/UserServiceAccessTest.java

```diff
@@ -170,131 +178,380 @@ public void test_updateUserWithRetries_register_includes_others() {
-        userAccessTierDao.getByUserAndAccessTier(dbUser, registeredTier);
-    assertThat(userAccessMaybe).isPresent();
-    assertThat(userAccessMaybe.get().getTierAccessStatusEnum())
-        .isEqualTo(TierAccessStatus.DISABLED);
-  }
-
-  @Test
-  public void testSimulateUserFlowThroughRenewal() {
-    providedWorkbenchConfig.access.enableAccessRenewal = true;
-    providedWorkbenchConfig.access.enableDataUseAgreement = true;
-    providedWorkbenchConfig.accessRenewal.expiryDays = (long) 365;
-    DbUser dbUser = new DbUser();
     dbUser.setDataUseAgreementSignedVersion(userService.getCurrentDuccVersion());
-
-    // The user is still compliant and has bypass set
     dbUser.setDataUseAgreementCompletionTime(
         Timestamp.from(
             START_INSTANT.minus(
                 (providedWorkbenchConfig.accessRenewal.expiryDays - 1), ChronoUnit.DAYS)));
-    dbUser = userService.updateUserWithRetries(registerUserNow, dbUser, Agent.asUser(dbUser));
-
-    assertThat(userAccessTierDao.findAll()).hasSize(1);
-    assertThat(userAccessTierDao.getAllByUser(dbUser).get(0).getTierAccessStatusEnum())
-        .isEqualTo(TierAccessStatus.ENABLED);
+    dbUser = userService.updateUserWithRetries(this::removeDuaBypass, dbUser, Agent.asUser(dbUser));
 
-    // User is compliant and we remove dua bypass
-    dbUser =
-        userService.updateUserWithRetries(
-            this::registerUserWithoutDuaBypass, dbUser, Agent.asUser(dbUser));
-
-    assertThat(userAccessTierDao.findAll()).hasSize(1);
-    assertThat(userAccessTierDao.getAllByUser(dbUser).get(0).getTierAccessStatusEnum())
-        .isEqualTo(TierAccessStatus.ENABLED);
+    // User is compliant
+    assertRegisteredTierEnabled(dbUser);
 
     // Simulate time passing, user is no longer compliant
     PROVIDED_CLOCK.setInstant(START_INSTANT.plus(2, ChronoUnit.DAYS));
-    dbUser =
-        userService.updateUserWithRetries(
-            this::registerUserWithoutDuaBypass, dbUser, Agent.asUser(dbUser));
-    assertThat(userAccessTierDao.getAllByUser(dbUser)).hasSize(1);
-    assertThat(userAccessTierDao.getAllByUser(dbUser).get(0).getTierAccessStatusEnum())
-        .isEqualTo(TierAccessStatus.DISABLED);
+    dbUser = userService.updateUserWithRetries(Function.identity(), dbUser, Agent.asUser(dbUser));
+    assertRegisteredTierDisabled(dbUser);
 
     // Simulate user filling out DUA, becoming compliant again
     dbUser.setDataUseAgreementCompletionTime(
         Timestamp.from(START_INSTANT.plus(2, ChronoUnit.DAYS)));
-    dbUser =
-        userService.updateUserWithRetries(
-            this::registerUserWithoutDuaBypass, dbUser, Agent.asUser(dbUser));
-    assertThat(userAccessTierDao.getAllByUser(dbUser)).hasSize(1);
-    assertThat(userAccessTierDao.getAllByUser(dbUser).get(0).getTierAccessStatusEnum())
-        .isEqualTo(TierAccessStatus.ENABLED);
+    dbUser = userService.updateUserWithRetries(Function.identity(), dbUser, Agent.asUser(dbUser));
```

Comment:
opt: existing issue, but maybe set dua within the callback for a slightly better simulation of the real flow